hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieUnMergedLogRecordScanner.java

```diff
@@ -135,10 +137,15 @@ public Builder withLogRecordScannerCallback(LogRecordScannerCallback callback) {
       return this;
     }
 
+    public Builder withInternalSchema(InternalSchema internalSchema) {
+      this.internalSchema = internalSchema;
+      return this;
```

Comment:
There is already a read schema, why we pass around another schema, whatever it is, please use just one schema !

Reply:
I used the second schema here to be consistent with `HoodieMergedLogRecordScanner` which already use this approach to scan logs in `HoodieMergeOnReadRDD#scanLog`

Reply:
Yes, a very confusing practice, the reader/format should be deterministic to one static given schema, it should not care about how the schema is generated, or where it comes from, say: it should not be imposed to the evolution logic.

Reply:
I reverted changes in `HoodieMergedLogRecordScanner`. Now there is only one schema -- `InternalSchema` which wraps `org.apache.avro.Schema`. The same approach is used in `HoodieUnMergedLogRecordScanner`